code/game/objects/items/devices/radio/radio.dm

```diff
@@ -743,16 +748,37 @@ var/global/list/default_interrogation_channels = list(
 	var/turf/T = get_turf(src)
 	var/obj/effect/overmap/visitable/V = GLOB.map_sectors["[T.z]"]
 	if(istype(V) && V.comms_support)
-		frequency = assign_away_freq(V.name)
+		var/freq_name = V.name
+		if(V.freq_name)
+			freq_name = V.freq_name
+		frequency = assign_away_freq(freq_name)
+		default_frequency = frequency
 		channels += list(
-			V.name = TRUE,
+			freq_name = TRUE,
 			CHANNEL_HAILING = TRUE
 		)
+		internal_channels += list(
+			num2text(frequency),
+			num2text(HAIL_FREQ)
+		)
 		if(V.comms_name)
 			name = "[V.comms_name] shortwave radio"
 
 	return ..()
 
+/obj/item/device/radio/hailing
+	bypass_default_internal = TRUE
+	default_frequency = HAIL_FREQ
+
+/obj/item/device/radio/hailing/Initialize()
+	channels = list(
+		CHANNEL_HAILING = TRUE
+	)
+	internal_channels += list(
+		num2text(HAIL_FREQ),
+	)
+	return ..()
```

Comment:
Internal channels are initialized in /obj/item/device/radio/Initialize(), which would override whatever you set here, you should call the parent init before whatever you do unless there's a very good reason not to (also for signals handling and other reasons